ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/diffbased/verkle/VerkleAccount.java

```diff
@@ -94,48 +94,58 @@ public static VerkleAccount fromRLP(
     final RLPInput in = RLP.input(encoded);
     in.enterList();
 
-    final long nonce = in.readLongScalar();
     final Wei balance = Wei.of(in.readUInt256Scalar());
-    final Hash storageRoot = Hash.wrap(in.readBytes32());
+    final long nonce = in.readLongScalar();
     final Hash codeHash = Hash.wrap(in.readBytes32());
-
+    final long codeSize = in.readLongScalar();
     in.leaveList();
 
     return new VerkleAccount(
-        context, address, address.addressHash(), nonce, balance, storageRoot, codeHash, mutable);
+        context, address, address.addressHash(), nonce, balance, codeSize, codeHash, mutable);
+  }
+
+  @Override
+  public Optional<Long> getCodeSize() {
+    return Optional.of(codeSize);
+  }
+
+  @Override
+  public void setCode(final Bytes code) {
+    super.setCode(code);
+    if (code == null || code.isEmpty()) {
+      this.codeSize = 0L;
+    } else {
+      this.codeSize = code.size();
+    }
+  }
+
+  @Override
+  public Hash getStorageRoot() {
+    // Verkle accounts do not have a storage root.
+    // For now, a Verkle account will always return EMPTY for this value,
+    // but this will need to be addressed in a future refactor.
+    return Hash.EMPTY_TRIE_HASH;
   }
 
   @Override
   public NavigableMap<Bytes32, AccountStorageEntry> storageEntriesFrom(
       final Bytes32 startKeyHash, final int limit) {
-    return ((BonsaiWorldStateKeyValueStorage) context.getWorldStateStorage())
-        .storageEntriesFrom(this.addressHash, startKeyHash, limit);
+    throw new RuntimeException(
+        "The method storageEntriesFrom is not supported when using Verkle tries. Verkle tries manage storage differently");
   }
 
   @Override
   public void writeTo(final RLPOutput out) {
     out.startList();
 
-    out.writeLongScalar(nonce);
     out.writeUInt256Scalar(balance);
-    out.writeBytes(storageRoot);
+    out.writeLongScalar(nonce);
```

Comment:
why did you swap nonce with balance for both reading and writing? Any reason in particular?

Reply:
yes it was to be compatible with the VerkleStateTrieAccountValue encoding part as I decided to implement it like that. no more reason than that I think I can revert this change